### ISSUE TITLE
`asyncio.Queue.task_done` documentation is ambiguous

### ISSUE THREAD

**Issue by enok71** [python/cpython#124433]
# Documentation

When reading https://docs.python.org/3/library/asyncio-queue.html the reader gets no clue to the fact that this object is (primarily) meant to contain tasks. (Because it is, isn't it? The task_done() method seems to indicate that)

### Linked PRs
* gh-128669
* gh-128671
* gh-128672

**Comment by ZeroIntensity** [comment]
No, an `asyncio.Queue` is just the asynchronous analogue of `queue.Queue`. It can hold any object, not just tasks. In this case, `task_done` is just badly named -- it's not really related to `asyncio.Task`, it's the equivalent of [Queue.task_done](https://docs.python.org/3/library/queue.html#queue.Queue.task_done).

I think this is a good indicator that the excerpt under `asyncio.Queue.task_done` should get reworded, the term "task" in the context of asyncio refers to a future, not a thread doing something to a queue.

**Comment by enok71** [comment]
In [here](https://docs.python.org/3/library/asyncio.html) it looks like the asyncio.queue is intended to be a "high-level API" to "distribute tasks"

**Comment by ZeroIntensity** [comment]
You can do that, but that isn't their sole purpose.

**Comment by JelleZijlstra** [comment]
"Task" here generally means "a thing to do", not "`asyncio.Task`".

**Comment by ZeroIntensity** [comment]
Yeah, that's why it's confusing in the context of `asyncio.Queue`

**Comment by enok71** [comment]
Wouldn't hurt to say something on the subject in the docs then.

**Comment by ZeroIntensity** [comment]
Would you like to author the PR? (If not, I can deal with it.)

**Comment by enok71** [comment]
Feel free. I don't know the subject well enough.
